packages/asset-value/src/assetValue.ts

```diff
@@ -0,0 +1,212 @@
+import { AssetId } from '@shapeshiftoss/caip'
+import { BigNumber } from 'bignumber.js'
+import { Md5 } from 'ts-md5'
+
+import { CHECKSUM_LENGTH, DEFAULT_FORMAT_DECIMALS } from './constants'
+import { AssetValueParams, isAssetValueParams, SerializedAssetValue } from './types'
+import { AssetValueFormat } from './types'
+import { bn, bnOrZero } from './utils'
+
+type AssetValueState = {
+  assetId: AssetId
+  precision: number
+  value: string
+}
+
+//TODO(pastaghost): Add JSDoc documentation
+export class AssetValue {
+  state: AssetValueState
```

Comment:
reason for object over standard class vars? also should probably make private to avoid mutation outside of class